packages/contracts-bedrock/test/L2/L2StandardBridge.t.sol

```diff
@@ -535,46 +754,167 @@ contract L2StandardBridge_Bridge_Test is Bridge_Initializer {
-    }
+        vm.expectCall(
+            address(l2CrossDomainMessenger),
+            500,
+            abi.encodeWithSelector(CrossDomainMessenger.sendMessage.selector, address(l1StandardBridge), message, 50000)
+        );
+
+        vm.expectEmit(address(l2StandardBridge));
+        emit ETHBridgeInitiated(alice, alice, 500, hex"dead");
+
+        // SentMessage event emitted by the CrossDomainMessenger
+        vm.expectEmit(address(l2CrossDomainMessenger));
+        emit SentMessage(address(l1StandardBridge), address(l2StandardBridge), message, nonce, 50000);
+
+        // SentMessageExtension1 event emitted by the CrossDomainMessenger
+        vm.expectEmit(address(l2CrossDomainMessenger));
+        emit SentMessageExtension1(address(l2StandardBridge), 500);
 
-    /// @dev Tests that bridging ETH reverts if a custom gas token is used.
-    function test_bridgeETHTo_isCustomGasToken_reverts() external {
         vm.prank(alice, alice);
+
+        l2StandardBridge.bridgeETH{ value: 500 }(50000, hex"dead");
+    }
+
+    /// @dev Tests that bridging succeeds with custom gas token.
+    function test_bridgeETH_customGasToken_succeeds() external {
         vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
-        vm.expectRevert("StandardBridge: cannot bridge ETH with custom gas token");
 
-        l2StandardBridge.bridgeETHTo(bob, 60000, hex"dead");
+        uint256 nonce = l2CrossDomainMessenger.messageNonce();
+
+        bytes memory message =
+            abi.encodeWithSelector(StandardBridge.finalizeBridgeETH.selector, alice, alice, 0, hex"dead");
+
+        vm.expectCall(
+            address(l2StandardBridge), 0, abi.encodeWithSelector(l2StandardBridge.bridgeETH.selector, 50000, hex"dead")
+        );
+
+        vm.expectCall(
+            address(l2CrossDomainMessenger),
+            0,
+            abi.encodeWithSelector(CrossDomainMessenger.sendMessage.selector, address(l1StandardBridge), message, 50000)
+        );
+
+        vm.expectEmit(address(l2StandardBridge));
+        emit ETHBridgeInitiated(alice, alice, 0, hex"dead");
+
+        // SentMessage event emitted by the CrossDomainMessenger
+        vm.expectEmit(address(l2CrossDomainMessenger));
+        emit SentMessage(address(l1StandardBridge), address(l2StandardBridge), message, nonce, 50000);
+
+        // SentMessageExtension1 event emitted by the CrossDomainMessenger
+        vm.expectEmit(address(l2CrossDomainMessenger));
+        emit SentMessageExtension1(address(l2StandardBridge), 0);
+
+        vm.prank(alice, alice);
+
+        l2StandardBridge.bridgeETH(50000, hex"dead");
     }
 
-    /// @dev Tests that bridging ETH reverts if a custom gas token is used.
-    function test_bridgeETH_isCustomGasToken_reverts() external {
+    /// @dev Tests that bridging reverts when value is passed with custom gas token.
+    function test_bridgeETH_customGasToken_withValue_reverts() external {
```

Comment:
You can delete above test if you turn this into a fuzz test because it will do the 0 case